airbyte-integrations/connectors/source-xero/source_xero/streams.py

```diff
@@ -0,0 +1,242 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import decimal
+import re
+from abc import ABC
+from datetime import date, datetime, time, timedelta, timezone
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+def parse_date(value):
+    # Xero datetimes can be .NET JSON date strings which look like
+    # "/Date(1419937200000+0000)/"
+    # https://developer.xero.com/documentation/api/requests-and-responses
+    pattern = r"Date\((\-?\d+)([-+])?(\d+)?\)"
+    match = re.search(pattern, value)
+
+    iso8601pattern = r"((\d{4})-([0-2]\d)-0?([0-3]\d)T([0-5]\d):([0-5]\d):([0-6]\d))"
+
+    if not match:
+        iso8601match = re.search(iso8601pattern, value)
+        if iso8601match:
+            try:
+                return datetime.strptime(value)
+            except Exception:
+                return None
+        else:
+            return None
+
+    millis_timestamp, offset_sign, offset = match.groups()
+    if offset:
+        if offset_sign == "+":
+            offset_sign = 1
+        else:
+            offset_sign = -1
+        offset_hours = offset_sign * int(offset[:2])
+        offset_minutes = offset_sign * int(offset[2:])
+    else:
+        offset_hours = 0
+        offset_minutes = 0
+
+    return datetime.fromtimestamp((int(millis_timestamp) / 1000), tz=timezone.utc) + timedelta(hours=offset_hours, minutes=offset_minutes)
+
```

Comment:
Can you add unit test for this function, also looks this function doesn't two different things, maybe you can break into two.

Reply:
IMO, it's not necessary to split these function in two as it's used for one certain thing: parse .NET JSON date format into `datetime` and these functions will never be used separately

I've added a unit test for this one, thanks for the catch